Dockerfile

```diff
@@ -1,5 +1,6 @@
-ARG AMAZONLINUX_VERSION=2.0.20221210.1-amd64
-FROM public.ecr.aws/amazonlinux/amazonlinux:${AMAZONLINUX_VERSION} as base
+ARG AMAZONLINUX_VERSION=2.0.20221210.1
+ARG ARCH=amd64
```

Comment:
IMO you can just use `$TARGETARCH` (which is provided by Docker), to eliminate the custom argument

See https://docs.docker.com/desktop/extensions-sdk/extensions/multi-arch/#adding-multi-arch-binaries

Reply:
I think that's just for extensions. [Maybe you're thinking of `BUILDPLATFORM` and `TARGETPLATFORM`?](https://docs.docker.com/build/building/multi-platform/#building-multi-platform-images)